hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java

```diff
@@ -357,6 +357,10 @@ public boolean deleteObject(String keyName, boolean recursive)
       bucket.deleteDirectory(keyName, recursive);
       return true;
     } catch (OMException ome) {
+      if (OMException.ResultCodes.KEY_NOT_FOUND == ome.getResult()) {
```

Comment:
Also please add it to BasicRootedOzoneClientAdapterImpl

Reply:
Looks like OFS does a getFileStatus call first irrespective of FSO or not. Ideally we should keep the behaviour same across ofs & o3fs . I'm okay with this fix too.

Reply:
For 03fs, getFileStatus is not added as it will impact performance, as for FSO, it will go iterative to identify root dir/file, So this is just to avoid another call and avoid performance impact, this is handled using Exception case to have warning.
ofs is existing behavior and not planning to change.